controllers/bounce_processes.go

```diff
@@ -192,22 +192,14 @@ func getProcessesReadyForRestart(logger logr.Logger, cluster *fdbv1beta2.Foundat
 		return nil, &requeue{message: "Waiting for config map to sync to all pods", delayedRequeue: true}
 	}
 
-	counts, err := cluster.GetProcessCountsWithDefaults()
-	if err != nil {
-		return nil, &requeue{
-			curError: err,
+	// If we upgrade the cluster wait until the minimum number of required processes are ready for the restart.
```

Comment:
Can't this cause the cluster to become unavailable (what if the processes that get left out are in a state like SidecarUnreachable then those processes will keep running at the older version and if their count exceeds the replication factor then reads won't succeed)?